Expand doc on deal-fitlering via external hooks (#567)
filecoin-project/filecoin-docs: docs/mine/lotus/miner-configuration.md

The final value of `ExpectedSealDuration` should equal `(TIME_TO_SEAL_A_SECTOR + WaitDealsDelay) * 1.5`. This equation ensures that the miner does not commit to having the sector sealed too soon.
:::

## Using filters for fine-grained Storage and Retrieval Deal acceptance

In certain situations it is desirable to have more precise and dynamic control over a combination of deal parameters like for example:
- cryptographic identity of client
- time of day / resource utilization
- content

> - etc.

Author: Mmmmm actually in most cases *it is not*. Steering folks enabling these by default would be a mistake in my eye. Can we find a middle ground?

> In certain situations it is desirable to have more precise and dynamic control over a combination of deal parameters like for example:
> - cryptographic identity of client
> - time of day / resource utilization
> - content
> - etc.

Reviewer: ```suggestion
Your use-case might demand very precise and dynamic control over a combination of deal parameters.
```

:::

## Using filters for fine-grained Storage and Retrieval Deal acceptance

In certain situations it is desirable to have more precise and dynamic control over a combination of deal parameters like for example:
- cryptographic identity of client
- time of day / resource utilization
- content
- etc.

Lotus provides two IPC hooks allowing you to name a command to execute for
every deal before it is accepted by the miner: `Filter` for storage deals and `RetrievalFilter` for retrieval deals. The executed command receives a JSON representation of the deal parameters on standard input, and is expected to
either exit with 0 (success, proceed with deal) or with non-0 (failure, reject deal).

- The most trivial filter rejecting any retrieval deal would be something like:
`RetrievalFilter = "/bin/false"`

- [This Perl script](https://gist.github.com/ribasushi/53b7383aeb6e6f9b030210f4d64351d5/9bd6e898f94d20b50e7c7586dc8b8f3a45dab07c#file-dealfilter-pl) is another example. It lets the miner deny specific clients, and to only accept deals that are set to start relatively soon.

> - You could also rely on a third party filtering framework like `bitscreen`

Reviewer 2: Can we instead say: 

You can also use a third party content policy framework like `bitscreen` by Murmuration Labs

Author: https://github.com/filecoin-project/filecoin-docs/pull/567/commits/02c75e99be36be7c6d5159f5957020cb7c660dd8
